eos-tech-support/eos-download-image

```diff
@@ -173,12 +177,30 @@ class EosDownloadImage(object):
 
         return keyring
 
-    def download_and_verify(self, meta, what):
+    def download_and_verify(self, meta, what, url=None, size=None):
+        """ Download and verify image. Caller should either use meta or specify url and size."""
         destdir = self.args.outdir
-        image_url = '{}/{}'.format(self.base, meta['file'])
-        image_size = meta['compressed_size']
-        image_mtime = dateutil.parser.parse(meta['last_modified'])
-        signature_url = '{}/{}'.format(self.base, meta['signature'])
+        if meta is not None:
+            image_url = '{}/{}'.format(self.base, meta['file'])
+            image_size = meta['compressed_size']
+            image_mtime = dateutil.parser.parse(meta['last_modified'])
```

Comment:
This would be:

```python
if 'last_modified' in meta:
    image_mtime = dateutil.parser.parse(meta['last_modified'])
else:
    image_mtime = None
```